src/EditorFeatures/CSharpTest/UseExpressionBody/Analyzer/UseExpressionBodyForPropertiesAnalyzerTests.cs

```diff
@@ -365,5 +365,64 @@ int Foo(int i)
 }", ignoreTrivia: false,
     options: UseBlockBody);
         }
+
+        [WorkItem(20362, "https://github.com/dotnet/roslyn/issues/20362")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseExpressionBody)]
+        public async Task TestOfferToConvertToBlockEvenIfExpressionBodyPreferredIfHasThrowExpressionPriorToCSharp7()
+        {
+            await TestAsync(
+@"
+using System;
+class C
+{
+    int Foo [|=>|] throw new NotImplementedException();
+}",
+@"
+using System;
+class C
+{
+    int Foo 
+    {
+        get
+        {
+            throw new NotImplementedException();
+        }
+    }
+}", options: UseExpressionBody, parseOptions: CSharpParseOptions.Default.WithLanguageVersion(LanguageVersion.CSharp6));
+        }
+
+        [WorkItem(20362, "https://github.com/dotnet/roslyn/issues/20362")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseExpressionBody)]
+        public async Task TestOfferToConvertToBlockEvenIfExpressionBodyPreferredIfHasThrowExpressionPriorToCSharp7_FixAll()
+        {
+            await TestAsync(
+@"
+using System;
+class C
+{
+    int Foo {|FixAllInDocument:=>|} throw new NotImplementedException();
+    int Bar => throw new NotImplementedException();
```

Comment:
:bulb: Add a third property to verify that the fix all only targets the error conditions:

```csharp
int StaysAsIs => 0;
```